lib/commons/dom/create-grid.js

```diff
@@ -254,65 +254,61 @@ function isFlexOrGridContainer(vNode) {
  * zIndex values for each stacking context parent.
  * @param {VirtualNode} vNode
  * @param {VirtualNode} parentVNode
- * @param {Number} nodeIndex
+ * @param {Number} index
  * @return {Number[]}
  */
-function createStackingOrder(vNode, parentVNode, nodeIndex) {
+function createStackingOrder(vNode, parentVNode, index) {
```

Comment:
I took a stab at refactoring this. Since you have a couple of repeated `stackingOrder.push(createContext())` calls I pulled finding that order value into its own function. I think this helps explain what's going on.

```js
function createStackingOrder(vNode, parentVNode, gridIndex) {
  const stackingOrder = parentVNode._stackingOrder.slice();

  // if an element creates a stacking context, find the first
  // true stack (not a "fake" stack created from positioned or
  // floated elements without a z-index) and create a new stack at
  // that point (step #5 and step #8)
  // @see https://www.w3.org/Style/css2-updates/css2/zindex.html
  if (isStackingContext(vNode, parentVNode)) {
    const index = stackingOrder.findIndex(({ value }) =>
      [ROOT_ORDER, FLOAT_ORDER, POSITION_ORDER].includes(value)
    );
    if (index !== -1) {
      stackingOrder.splice(index, stackingOrder.length - index);
    }
  }

  const stackOrderValue = getStackOrderValue(vNode, parentVNode);
  if (stackOrderValue !== null) {
    stackingOrder.push(createContext(stackOrderValue, vNode, gridIndex));
  }
  return stackingOrder;
}

function getStackOrderValue(vNode, parentVNode) {
  const zIndex = getRealZIndex(vNode, parentVNode);
  if (!['auto', '0'].includes(zIndex)) {
    return parseInt(zIndex);
  }
  // if a positioned element has z-index: auto or 0 (step #8), or if
  // a non-positioned floating element (step #5), treat it as its
  // own stacking context @see https://www.w3.org/Style/css2-updates/css2/zindex.html
  if (vNode.getComputedStylePropertyValue('position') !== 'static') {
    return POSITION_ORDER;
  }
  // Put floated elements above z-index: 0
  // (step #5 floating get sorted below step #8 positioned)
  if (vNode.getComputedStylePropertyValue('float') !== 'none') {
    return FLOAT_ORDER;
  }
  if (isStackingContext(vNode, parentVNode)) {
    return DEFAULT_ORDER;
  }
  return null;
}
```